vfox-restServer/src/main/java/com/vfoexchange/restServer/serviceImpl/UserServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.vfoexchange.restServer.serviceImpl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import com.vfoexchange.restServer.dao.UserDao;
+import com.vfoexchange.restServer.model.User;
+import com.vfoexchange.restServer.service.UserService;
+
+@Service("userService")
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    UserDao userDao;
```

Comment:
@Transactional can be applied at class level or method level. Applying it at class level means all the public methods defined in the class are transactional.

Reply:
Hmm... I wasn't talking about '@transactional'. I was talking about using 'UserDao' as an 'instance level' variable. This can be thread-safe only if UserServiceImpl is NOT used as a Singleton. If it's used as a Singleton then it won't work in multi-threading environment. Anyway, we can tackle this later.